src/hotspot/share/opto/macroArrayCopy.cpp

```diff
@@ -169,6 +170,130 @@ void PhaseMacroExpand::generate_limit_guard(Node** ctrl, Node* offset, Node* sub
   generate_guard(ctrl, bol_lt, region, PROB_MIN);
 }
 
+//
+// Partial in-lining handling for smaller conjoint/disjoint array copies having
+// length(in bytes) less than ArrayCopyPartialInlineSize.
+//  if (length <= ArrayCopyPartialInlineSize) {
+//    partial_inlining_block:
+//      mask = Mask_Gen
+//      vload = LoadVectorMasked src , mask
+//      StoreVectorMasked dst, mask, vload
+//  } else {
+//    stub_block:
+//      callstub array_copy
+//  }
+//  exit_block:
+//    Phi = label partial_inlining_block:mem , label stub_block:mem (filled by caller)
+//    mem = MergeMem (Phi)
+//    control = stub_block
+//
+//  Exit_block and associated phi(memory) are partially initialized for partial_in-lining_block
+//  edges. Remaining edges for exit_block coming from stub_block are connected by the caller
+//  post stub nodes creation.
+//
+//  Constant copy length operation having size less than ArrayCopyPartialInlineSize prevents
+//  creation of additional control flow for stub_block and exit_block.
+//
+//  Return true to emit subsequent stub calling code.
+//
+
+bool PhaseMacroExpand::generate_partial_inlining_block(Node** ctrl, MergeMemNode** mem, const TypePtr* adr_type,
+                                                       RegionNode** exit_block, Node** result_memory, Node* length,
+                                                       Node* src_start, Node* dst_start, BasicType type) {
+  const TypePtr *src_adr_type = _igvn.type(src_start)->isa_ptr();
+
+  Node* orig_mem = *mem;
+  Node* is_lt64bytes_tp = NULL;
+  Node* is_lt64bytes_fp = NULL;
```

Comment:
It is difficult distinguish between _tp and _fp. Please, use whole words: false, true.